backend/packages/Upgrade/src/api/services/SegmentService.ts

```diff
@@ -377,7 +378,9 @@ export class SegmentService {
     return queryBuilder;
   }
 
-  public upsertSegment(segment: SegmentInputValidator, logger: UpgradeLogger): Promise<Segment> {
+  public async upsertSegment(segment: SegmentInputValidator, logger: UpgradeLogger): Promise<Segment> {
+    await this.checkIsDuplicateSegmentName(segment.name, segment.context, logger);
```

Comment:
This method is also used when editing - so now editing always fails unless you're changing the name.